### PR TITLE
[KOGITO-3922] Adding error workflow example

### PR DESCRIPTION
**Pull request by fjtirado**
Merge with https://github.com/kiegroup/kogito-runtimes/pull/1536

### CODE REVIEW

File: serverless-workflow-error-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+This example contains a simple greeting workflow service that illustrate error handling. 
+The service is described using JSON format as defined in the 
+[CNCF Serverless Workflow specification](https://github.com/cncf/wg-serverless/tree/main/workflow/spec).
+
+The workflow check if number is odd or event and print a message indicating that. 
+The main feature of this demo is that if the number is odd, an exception is thrown and it is the exception error handling the one that sets the odd message. 
+
+Hence, this workflow expects as JSON input containing a natural number. 
+
+The workflow starts with a
```
Comment (ricardozanini) [inline serverless-workflow-error-quarkus/README.md]: with a... ? :)

File: serverless-workflow-error-quarkus/README.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Kogito Serverless Workflow - Greeting Example
+
+## Description
+
+This example contains a simple greeting workflow service that illustrate error handling. 
+The service is described using JSON format as defined in the 
+[CNCF Serverless Workflow specification](https://github.com/cncf/wg-serverless/tree/main/workflow/spec).
```
Comment (tiagodolphine) [inline serverless-workflow-error-quarkus/README.md]: the link is broken

File: serverless-workflow-error-quarkus/README.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+The workflow check if number is odd or event and print a message indicating that. 
+The main feature of this demo is that if the number is odd, an exception is thrown and it is the exception error handling the one that sets the odd message. 
+
+Hence, this workflow expects as JSON input containing a natural number.
```
Comment (tiagodolphine) [inline serverless-workflow-error-quarkus/README.md]: maybe it is worth adding a link os some explanation to show how to define errors, like https://github.com/serverlessworkflow/specification/blob/main/specification.md#workflow-error-handling

**Review by tiagodolphine — APPROVED** [review]

File: serverless-workflow-error-quarkus/README.md (hunk 1 above)
Comment (fjtirado) [reply serverless-workflow-error-quarkus/README.md]: a wondeful description of the workflow which I forgot to add but now you can read ;)

File: serverless-workflow-error-quarkus/README.md (hunk 2 above)
Comment (fjtirado) [reply serverless-workflow-error-quarkus/README.md]: sure, copy paste issue, fixed

File: serverless-workflow-error-quarkus/README.md (hunk 3 above)
Comment (fjtirado) [reply serverless-workflow-error-quarkus/README.md]: yes, I added it in the update readme, thanks a lot

File: serverless-workflow-error-quarkus/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# Kogito Serverless Workflow - Greeting Example
+
+## Description
+
+This example contains a simple greeting workflow service that illustrate error handling. 
+The service is described using JSON format as defined in the 
+[CNCF Serverless Workflow specification](https://github.com/serverlessworkflow/specification).
+
+The workflow check if number is odd or event and print a message indicating that.
```
Comment (ricardozanini) [inline serverless-workflow-error-quarkus/README.md]:
```suggestion
The workflow check if the number is odd or even and print a message indicating that. 
```

File: serverless-workflow-error-quarkus/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
+# Kogito Serverless Workflow - Greeting Example
+
+## Description
+
+This example contains a simple greeting workflow service that illustrate error handling. 
+The service is described using JSON format as defined in the 
+[CNCF Serverless Workflow specification](https://github.com/serverlessworkflow/specification).
+
+The workflow check if number is odd or event and print a message indicating that. 
+The main feature of this demo is that if the number is odd, an exception is thrown and it is the exception error handling the one that sets the odd message.
```
Comment (ricardozanini) [inline serverless-workflow-error-quarkus/README.md]:
```suggestion
The main feature of this demo is that if the number is odd, an exception is thrown, and it is the exception error handling the one that sets the odd message. 
```

File: serverless-workflow-error-quarkus/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+[CNCF Serverless Workflow specification](https://github.com/serverlessworkflow/specification).
+
+The workflow check if number is odd or event and print a message indicating that. 
+The main feature of this demo is that if the number is odd, an exception is thrown and it is the exception error handling the one that sets the odd message. 
+
+Hence, this workflow expects as JSON input containing a natural number. This number is passed using a service operation to ``EvenService`` java class. If the number is even, the workflow move to the next defined state, which injects "even" ``numberType``. But if the number is odd, the class throws an ``IllegalArgumentException``. This exception is handled and redirected to odd inject node by using [inline workflow error handling](https://github.com/serverlessworkflow/specification/blob/main/specification.md#Workflow-Error-Handling).  This basically consist on adding as ``onErrors`` field, where the expected exception is specified in ``code`` and the target state (a node injecting "odd" ``numberType``) in ``transition``. Both execution paths finish on the same node, which prints the calculated ``eventType``.
```
Comment (ricardozanini) [inline serverless-workflow-error-quarkus/README.md]:
```suggestion
Hence, this workflow expects JSON input containing a natural number. This number is passed using a service operation to `EvenService` java class. If the number is even, the workflow moves to the next defined state, injecting "even" `numberType`. But if the number is odd, the class throws an `IllegalArgumentException`. This exception is handled and redirected to odd inject node by using [inline workflow error handling](https://github.com/serverlessworkflow/specification/blob/main/specification.md#Workflow-Error-Handling).  This basically consists on adding `onErrors` field, where the expected exception is specified in `code` and the target state (a node injecting "odd" `numberType`) in `transition`. Finally, both execution paths finish on the same node, which prints the calculated `eventType`.
```

**Review by ricardozanini — APPROVED** [review]
Minor tweaks in the README. Thanks, man!

**Pull request merged**